### PR TITLE
doc: added docstrings for QdrantDocumentStore

### LINKED ISSUE #764: Add docstrings for `QdrantDocumentStore`
**Is your feature request related to a problem? Please describe.**
Api reference for `QdrantDocumentStore` is not found on https://docs.haystack.deepset.ai/reference/integrations-qdrant. I believe it's because there are no docstrings for it

**Describe the solution you'd like**
Add docstrings explaining params

**Describe alternatives you've considered**
N/A

**Additional context**
For reference, this is what WeaviateDocumentStore API ref looks like: https://docs.haystack.deepset.ai/reference/integrations-weaviate

### PR DESCRIPTION
**Pull request by Amnah199**
### Related Issues

- fixes #764 

### Proposed Changes:

Added the missing doc strings in QdrantDocumentStore class to generate documentation for API reference.

### How did you test it?

Existing unit tests

### Notes for the reviewer

NA
### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 1]
```diff
@@ -187,6 +320,15 @@ def filter_documents(
         self,
         filters: Optional[Union[Dict[str, Any], rest.Filter]] = None,
     ) -> List[Document]:
+        """
+        Returns the documents that match the provided filters.
+
+        For a detailed specification of the filters, refer to the
+        DocumentStore.filter_documents() protocol documentation.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:327]:
```suggestion
        For a detailed specification of the filters, refer to the
        [documentation](https://docs.haystack.deepset.ai/docs/metadata-filtering).
```

This seems more user-friendly.

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 2]
```diff
@@ -204,6 +346,20 @@ def write_documents(
...
+        """
+        Writes documents to Qdrant using the specified policy.
+        The QdrantDocumentStore can handle duplicate documents based on the given policy.
+        The available policies are:
+        - `FAIL`: The operation will raise an error if any document already exists.
+        - `OVERWRITE`: Existing documents will be overwritten with the new ones.
+        - `SKIP`: Existing documents will be skipped, and only new documents will be added.
+
+        This method validates the documents, sets up the collection, handles duplicates based on the policy,
+        and batches the documents for efficient upsertion into Qdrant.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:358]:
```suggestion

```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 3]
```diff
@@ -204,6 +346,20 @@ def write_documents(
...
+
+        :param documents: A list of Document objects to write to Qdrant.
+        :param policy: The policy for handling duplicate documents. Defaults to `FAIL`.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:361]:
```suggestion
        :param policy: The policy for handling duplicate documents.
        
        :returns: The number of documents written to the document store.        
```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 4]
```diff
@@ -239,6 +395,11 @@ def write_documents(
         return len(document_objects)
 
     def delete_documents(self, ids: List[str]):
+        """
+        Deletes all documents with matching document_ids from the DocumentStore.
+
+        :param document_ids: The object_ids to delete.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:401]:
```suggestion
        Deletes documents that match the provided `document_ids` from the document store.

        :param document_ids: the document ids to delete
```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 5]
```diff
@@ -271,6 +446,13 @@ def get_documents_generator(
         self,
         filters: Optional[Union[Dict[str, Any], rest.Filter]] = None,
     ) -> Generator[Document, None, None]:
+        """
+        Returns a generator that yields documents from Qdrant based on the provided filters.
+
+        :param filters: Filters applied to the retrieved documents.
+        :returns: A generator that yields Haystack Document objects retrieved from Qdrant.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:453]:
```suggestion
        :returns: A generator that yields documents retrieved from Qdrant.
```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 6]
```diff
@@ -299,6 +481,16 @@ def get_documents_by_id(
...
     ) -> List[Document]:
+        """
+        Retrieves documents from Qdrant by their IDs.
+
+        :param ids:
+            A list of document IDs to retrieve.
+        :param index:
+            The name of the index to retrieve documents from.
+        :returns:
+            A list of Haystack Document objects.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:492]:
```suggestion
            A list of documents.
```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 7]
```diff
@@ -325,6 +517,21 @@ def _query_by_sparse(
         scale_score: bool = True,
         return_embedding: bool = False,
     ) -> List[Document]:
+        """
+        Queries Qdrant using sparse embeddings and returns the most relevant documents.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:521]:
```suggestion
        Queries Qdrant using a sparse embedding and returns the most relevant documents.
```

(Nit: This is an internal method and probably did not need to be documented in depth. Better that way, anyway!)

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 8]
```diff
@@ -367,6 +574,17 @@ def _query_by_embedding(
         scale_score: bool = True,
         return_embedding: bool = False,
     ) -> List[Document]:
+        """
+        Queries Qdrant using dense embeddings and returns the most relevant documents.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:578]:
```suggestion
        Queries Qdrant using a dense embedding and returns the most relevant documents.
```

**Review by anakin87 — COMMENTED** [review]
Good job!

I found some (minor) opportunities for improvement.

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 9]
```diff
@@ -96,6 +135,97 @@ def __init__(
         scroll_size: int = 10_000,
         payload_fields_to_index: Optional[List[dict]] = None,
     ):
+        """
+        :param location:
+            If “:memory:” - use in-memory Qdrant instance. If str - use it as a url parameter.
+            If None - use default values for host and port.
+        :param url:
+            Either host or str of “Optional[scheme], host, Optional[port], Optional[prefix]”. Default: None.
```
[the remark this answers is not in the document]
Comment (Amnah199) [reply integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:143]: Noted, will update the rest.

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 10]
```diff
@@ -96,6 +134,98 @@ def __init__(
...
+        :param api_key:
+            API key for authentication in Qdrant Cloud.
+        :param prefix:
+            If not `None` - add prefix to the REST URL path.
+            Example: service/v1 will result in http://localhost:6333/service/v1/{qdrant-endpoint}
+            for REST API.
+        :param timeout:
+            Timeout for REST and gRPC API requests.
+        :param host:
+            Host name of Qdrant service. If url and host are None, set to `localhost`.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:161]:
```suggestion
            Host name of Qdrant service. If `url` and `host` are `None`, set to `localhost`.
```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 11]
```diff
@@ -96,6 +134,98 @@ def __init__(
...
+        :param on_disk:
+            Whether to store the collection on disk.
+        :param content_field:
+            The field for the document content.
+        :param name_field:
+            The field for the document name.
+        :param embedding_field:
+            The field for the document embeddings.
+        :param use_sparse_embedding:
+            If set to True, enables the use for 'Sparse Embedding' class.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:180]:
```suggestion
            If set to `True`, enables support for sparse embeddings.

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 12]
```diff
@@ -96,6 +134,98 @@ def __init__(
...
+        :param similarity:
+            The similarity metric to use.
+        :param return_embedding:
+            Whether to return embeddings in the search results.
+        :param progress_bar:
+            Whether to show a progress bar or not.
+        :param duplicate_documents:
+            The policy for handling duplicate documents ("overwrite", "skip", or "fail").
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:188]:
```suggestion
        :param duplicate_documents:
            This parameter is not used and will be removed in a future release .
```

I discovered this

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 13]
```diff
@@ -96,6 +134,98 @@ def __init__(
...
+            This setting saves RAM by slightly increasing response time.
+            Note: indexed payload values remain in RAM.
+        :param hnsw_config:
+            Params for HNSW index.
+        :param optimizers_config:
+            Params for optimizer.
+        :param wal_config:
+            Params for Write-Ahead-Log.
+        :param quantization_config:
+            Params for quantization. If None, quantization will be disabled.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:214]:
```suggestion
            Params for quantization. If `None`, quantization will be disabled.
```

File: integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py [hunk 14]
```diff
@@ -325,6 +516,21 @@ def _query_by_sparse(
         scale_score: bool = True,
         return_embedding: bool = False,
     ) -> List[Document]:
+        """
+        Queries Qdrant using sparse a embedding and returns the most relevant documents.
```
Comment (anakin87) [inline integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py:520]:
```suggestion
        Queries Qdrant using a sparse embedding and returns the most relevant documents.
```

**Review by anakin87 — APPROVED** [review]
Feel free to merge the PR once the requested changes are incorporated

**Pull request merged**